lib/versioner/systemtrash_darwin.go

```diff
@@ -0,0 +1,87 @@
+// Copyright (C) 2016 The Syncthing Authors.
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at http://mozilla.org/MPL/2.0/.
+
+//+build darwin,cgo
+
+package versioner
+
+/*
+#cgo CFLAGS: -x objective-c
+#cgo LDFLAGS: -framework Foundation
+#import <stdlib.h>
+#import <Foundation/Foundation.h>
+
+const char *trashFile(char *s) {
+	NSAutoreleasePool *pool = [NSAutoreleasePool new];
```

Comment:
(and you can also just use the autorelease'd `[NSURL fileURLWithPath:fileStr]`, instead of explicitly autoreleasing it with `[[[NSURL alloc] initFileURLWithPath:fileStr] autorelease]`)